src/DataTypes/Serializations/SerializationInfo.h

```diff
@@ -0,0 +1,96 @@
+#pragma once
+
+#include <Core/Block.h>
+#include <DataTypes/Serializations/ISerialization.h>
+
+namespace DB
+{
+
+/** Contains information about kinds of serialization of columns.
+ *  Also contains information about content of columns,
+ *  that helps to choose kind of serialization of column.
+ *
+ *  Currently has only information about number of default rows,
+ *  that helps to choose sparse serialization.
+ *
+ *  Should be extended, when new kinds of serialization will be implemented.
+ */
+class SerializationInfo
+{
+public:
+    using NameToKind = std::unordered_map<String, ISerialization::Kind>;
+
+    SerializationInfo() = default;
+    SerializationInfo(size_t number_of_rows_, const NameToKind & kinds);
+
+    static constexpr auto version = 1;
+    size_t getNumberOfDefaultRows(const String & column_name) const;
+    ISerialization::Kind getKind(const String & column_name) const;
+
+    size_t getNumberOfRows() const { return number_of_rows; }
+
+    void readText(ReadBuffer & in);
+    void writeText(WriteBuffer & out) const;
+
+    static NameToKind getKinds(const Block & block);
+    static NameToKind readKindsBinary(ReadBuffer & in);
+    static void writeKindsBinary(const NameToKind & kinds, WriteBuffer & out);
+
+private:
+    void fromJSON(const String & json_str);
+    String toJSON() const;
```

Comment:
Does this go to serialization.txt in data parts?

Reply:
Yes.